core/src/components/searchbar/test/basic/searchbar.e2e.ts

```diff
@@ -0,0 +1,37 @@
+import { expect } from '@playwright/test';
+import { test } from '@utils/test/playwright';
+
+test.describe('searchbar: basic', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto(`/src/components/searchbar/test/basic`);
+  });
+
+  test('should not have visual regressions', async ({ page }) => {
+    await page.setIonViewport();
+    expect(await page.screenshot()).toMatchSnapshot(`searchbar-diff-${page.getSnapshotSettings()}.png`);
+  });
+
+  test('should show cancel button on focus if show-cancel-button=focus', async ({ page }) => {
+    const searchbar = page.locator('#basic');
+
+    await searchbar.evaluate((el: HTMLIonSearchbarElement) => el.setFocus());
+    await page.waitForChanges();
+
+    expect(searchbar).toHaveClass(/searchbar-has-focus/);
```

Comment:
~ It may help to put `await` on these assertions, in an odd event that the `waitForChanges` threshold doesn't fully re-render with the applied classes. It will then await up to another 5000ms before failing.

Reply:
+1 `toHaveClass` returns a Promise, so we should be awaiting this